src/mesh/Router.cpp

```diff
@@ -316,15 +323,18 @@ bool perhapsDecode(meshtastic_MeshPacket *p)
         config.device.rebroadcast_mode == meshtastic_Config_DeviceConfig_RebroadcastMode_ALL_SKIP_DECODING)
         return false;
 
-    if (p->which_payload_variant == meshtastic_MeshPacket_decoded_tag)
```

Comment:
Is there any case in which we might have an already decoded a packet before we call perhapsDecode and this short circuit would still be desired?

Reply:
good catch. indeed, this can only run once.